src/worker.js

```diff
@@ -18,6 +18,13 @@ var ENVIRONMENT_IS_NODE = typeof process == 'object' && typeof process.versions
 if (ENVIRONMENT_IS_NODE) {
   // Create as web-worker-like an environment as we can.
 
+  // See the parallel code in shell.js.
+#if EXPORT_ES6 && ENVIRONMENT_MAY_BE_WEB
```

Comment:
I'm confused about `ENVIRONMENT_MAY_BE_WEB`.. surely `require` never works on the web does it?

Where does worker.js use other than below in the `ENVIRONMENT_MAY_BE_NODE` block?

It we only use require for those two lines below maybe we should replace them with dyanmic imports?

e.g.:

#if EXPORT_ES6
  var fs = await import('fs');                                                        
  var vm = await import('vm'); 
#else
   var fs = require('fs');                                                        
   var vm = require('vm'); 
#endif

Reply:
Yeah in the main JS `ENVIRONMENT_MAY_BE_WEB` is only used because there are static imports for when we know that environment is Node.js and no other.

In this case there are no static imports to fall back to, so it should either add them or just use `await import` like you suggested.

Reply:
Yeah, it's confusing. The comment this refers to explains more:

https://github.com/emscripten-core/emscripten/blob/7a14e5b2bea9dcf82d3e98ce1615544801183622/src/shell.js#L193-L196

My understanding of the lines is that in a pure node build we don't emit this anyhow. But in a mixed build we can't do that, and so if we build for node or web but end up running in node, then we get to this location, and need something to put here (since we couldn't put the static thing for a pure node build).

(This is all separate from this PR, of course, as it's preexisting.)

Reply:
It turns out I was wrong, this is not separable. This PR was enough to fix the case of using a `package.json` file to force ES6 module format, but without such a file node.js uses the suffix to decide what to run. And this PR adds `await import`, which errors in a non-ES6 module.

To fix that the PR now emits `.worker.mjs` when `EXPORT_ES6`. That is a breaking change unfortunately, which may break all users that copy files to a production environment that use `pthreads+EXPORT_ES6`, but I don't see a way around it? Hopefully not many are using `EXPORT_ES6`, and probably few are because of issues like this, actually...

Thoughts?

Reply:
I think if you use `EXPORT_ES6` when it makes sense to create worker.mjs.   I don't think think its a huge deal that we change this now.   Maybe just mention it in the changelog.

Reply:
Agreed, yeah, already added to the changelog.

If this breaking change sgty then let me know if you have any other feedback on the PR, otherwise I think it is done.